### DISCUSSION TITLE
Woche 04 - Testing

### DISCUSSION THREAD

**Discussion by lukashert** [Chrissi-Ruege/HealthKeeper#26] [General]
# Refactoring
Nachdem sich unsere Gruppe zu Beginn des Semesters verkleinert hat, haben wir uns letzte Woche dazu entschieden große Teile der Software strukturell zu verändern. Dadurch befindet sich die Software aktuell in keinem ausführbaren Stand.

# Testing
Zu Beginn wollen wir auf zwei Arten von Tests setzen: Manuelle Tests und Unit Tests.
Bei den Unit Tests sollte eine recht geringe Abdeckung von etwa 50% angestrebt werden. 
Diese reicht aus, da die Anwendung noch recht wenige Funktionen besitzt. ASP.NET MVC Apps bestehen vor allem aus drei Komponenten: Models, Views und Controller. Die Models und Views können kaum getestet werden, da sich darin keine Logik befindet. 
Da besonders UI-Tests eine große Herausforderung darstellen wird darauf verzichtet. 

Die Unit Tests werden mit NUnit umgesetzt und werden beim kompilieren der Anwendung ausgeführt.

Später wäre es möglich die Anwendung um Intergrationstests zu erweitern.

**Comment by ghost** [comment]
Spannendes Update!

Es liest sich so, als sei der prozentual angestrebte Wert der Testabdeckung an die aktuell vorhandene Funktionalität geknüpft. Habt ihr demnach vor, eure Testabdeckung im Projektverlauf weiter anzupassenn?

**Comment by JannisFuchs** [comment]
Das hört sich interessant an.
